### ISSUE TITLE
[Minor Issue] Press mold recipes could use a warning about consuming the gear/plate/rod in handbook and NEI/JEI.

### ISSUE THREAD

**Issue by MonaMoonlight** [BluSunrize/ImmersiveEngineering#1038]
~~I don't know if this is by design or not, but metal press molds are consuming the base item; in Tinker's Construct, where molten metal is used to form the mold, the base item is not destroyed in the process, so it would make sense if I didn't lose, say, a fairly expensive Electrum or Enderium gear when making a press mold.~~

~~If the way it currently works is by design, then I would suggest changing it, as you would not discard a gear used to make a mold. Same goes for a rod, or a plate.~~

~~**Edit:** I have since found out that TConstruct for 1.8.9, which IE currently doesn't work with, does destroy the base item, due to the temperature of the molten metal apparently being hot enough to do so. IE's recipe does not involve molten metal, however, and my suggestion still stands.~~

**Edit 2:** After @malte0811 explained to me why the crafting works as it currently does, I only have one thought; that there needs to be a note about the gear/plate/rod being consumed by the recipe, underneath its entry in the Engineer's Handbook, and under the recipe view in NEI/JEI, much like TiC has done in their 1.8.9 version.

**Comment by BluSunrize** [comment]
Meeeh. Not that important is it? Just don't use something super expensive. Use something cheap like a copper gear?

**Comment by MonaMoonlight** [comment]
Well, it's a bit like how lava buckets used to be consumed when used as furnace fuel, you know?
It didn't work out in the long run, because eventually you would've run out of material to make buckets from.
Given how much materials go into making the multiblock machines, it would make sense to conserve materials.

And like I said, it's not just gears, it's rods and plates too.

**Comment by laz2727** [comment]
a) Copper gears and iron rods and plates are cheap.
b) In TiC, base item *is* destroyed.

2016-03-30 12:56 GMT+05:00 MonaMoonlight <notifications@github.com>:

**Comment by laz2727** [comment]
a) Copper gears and iron rods and plates are cheap.
b) In TiC, base item _is_ destroyed.

**Comment by MonaMoonlight** [comment]
@laz2727 
a) they are fairly cheap, yes, but the loss adds up.
b) In Tinker's Construct, when making a mold with molten metal in a molding table, it does _not_ destroy the base item, it just pops right out, **as one would expect**. 
(Maybe you're using a different version, but in the latest one for 1.7.10 that I'm using, TConstruct-1.7.10-1.8.8, making a mold with molten aluminum brass does not destroy the base item.)

The problem is that it doesn't make any sense for the base item to be lost; if I was making a mold for anything that could be pressed/melted IRL, I wouldn't use a process that destroyed the original.
I mean, I might need to use it again, to make another mold, or I might like to use it in a machine.

**Comment by laz2727** [comment]
a) if you have a problem with losing one iron ingot per cast, don't play IE.
b) in latest TiC, casting casts does consume the casted part, which is
pretty logical considering you're pouring red hot metal on it.

**Comment by malte0811** [comment]
I just had a look at the code that would be required to do this, it is very ugly to do (it would be necessary to do this in the `onItemCrafted`-Event where you don't have access to the `IRecipe`-Object) and would not work reliably/at all with automated crafting (similar to how Ars Magica foci (or something like that) that required a certain level could be auto-crafted in an AE system without that level in 1.6). That alone is a reason for me not to implement this change.
And balance wise it should not be that bad of an impact. You don't really need more than 1, maybe 2, of each mold and they last forever. That will take 6 or 7 ingots of iron (1 for the rod, 1 for the plate and 4-5 for the gear) and 12 ingots of steel (but those 12 ingots aren't the subject of this discussion), which is not much compared to how much iron and steel everything takes (for example any IC2 machine takes 8 iron plates for the machine block plus some additional stuff).

**Comment by MonaMoonlight** [comment]
@malte0811 
Aha, so that's why... well, it makes sense now, thanks for explaining. In hindsight, I guess that may have been why the TiC creators made it so their mold crafting ruined the base item in the 1.8.9 version.

@laz2727 
a) It's 5 ingots per cast, one iron, and four of whatever material you wanted the cast to be.
b) Before you made the reply that I am now replying to, I did check the absolutely latest TiC, and yes, you're absolutely right, it does destroy the base item by way of red hot metal, but IE does not work with the same version of Minecraft as that version of TiC, so that is why I did not check that first. I was unaware, but I am not anymore. 

I can also see now from malte's comment, that it may have been necessary for balance between IE and TiC, so neither of them is better than the other at saving resources compared to standard crafting.
